src/mqtt.client.ts

```diff
@@ -82,16 +82,15 @@ export class MqttClient<
 
     protected keepAliveTimer?: TimerRef;
 
-    protected autoReconnect?: boolean | MqttAutoReconnectOptions;
-    protected reconnectAttempt = 1;
+    #reconnect?: MqttsReconnectStrategy;
```

Comment:
This should use typescripts visibility and a more descriptive name.

```suggestion
    private reconnectStrategy?: MqttsReconnectStrategy;
```